File: (9)jumpFloor/Solution.java
/*
题目描述

一只青蛙一次可以跳上1级台阶，也可以跳上2级。求该青蛙跳上一个n级的台阶总共有多少种跳法。
*/
public class Solution {

    public int JumpFloor(int target) {
    	if(target == 0)
    		return 0;
    	else if(target == 1)
    		return 1;
    	else if(target == 2)
    		return 2;
    	else if(target >2)
    		return JumpFloor(target-1)+JumpFloor(target-2);
    	return 0;
    }

    public static void main(String [] args){
    	Solution s = new Solution();
    	System.out.println(s.JumpFloor(-1));
    }

}

/*
1.第一步如何跳
2.进行下一步的递归
*/

File: (9)jumpFloor/Solution.cpp
class Solution {
public:
    int jumpFloor(int number) {
        if(number < 1)
        	return 0;
        if(number == 1)
        	return 1;
        if(number == 2)
        	return 2;
        return jumpFloor(number -1) + jumpFloor(number - 2);
    }

    int jumpFloor(int number) {
    	if(number < 1)
    		return 0;
    	int num1 = 1;
    	int num2 = 2;
    	if (number == 1)
    		return num1;
    	if (number == 2)
    		return num2;
    	int index = 3;
    	int result = 0;
    	while(index <= number) {
    		result = num1 + num2;
    		num1 = num2;
    		num2 = result;
    		index++;
    	}
    	return result; 
    }
};